src/hotspot/share/classfile/verifier.cpp

```diff
@@ -1675,7 +1675,7 @@ void ClassVerifier::verify_method(const methodHandle& m, TRAPS) {
         case Bytecodes::_return :
           if (return_type != VerificationType::bogus_type()) {
             verify_error(ErrorContext::bad_code(bci),
-                         "Method expects a return value");
+                         "Method descriptor has a non-void return type");
```

Comment:
I prefer the original text. The new text makes me have to think about what the actual problem is too much.